### PR TITLE
MGDAPI-5808 Removing OO

### PR DESCRIPTION
**Pull request by Boomatang**
# Issue link

https://issues.redhat.com/browse/MGDAPI-5808

# What

As part of the move to OBO the observability operator and all its components can be removed.
This PR work to remove OO, there is need to be followup work to remove the code required to remove OO.

# Verification steps

## NOTE
Uninstallation is broken and needs to be manually cleaned up. This issue is as a result of https://github.com/integr8ly/integreatly-operator/pull/3238 and a follow up JIRA is being created. This issue is being addressed in https://issues.redhat.com/browse/MGDAPI-5866

### Indexes
**Current Feature:** quay.io/jfitzpat/managed-api-service-index:1.38.0-5808
**Updated Version:** quay.io/jfitzpat/managed-api-service-index:1.39.0-5808 (non service affecting)

## Fresh installations
* On a cluster that has never had RHOAM installed, Install the index `Updated Version`.
  * Ensure the latest config image is used from managed tenants bundles.
* Install should complete
  * Observability stage in the RHMI CR should be complete
  * No log errors over the observability kind should be in the RHOAM logs
  * The observability operator should not be installed

## Upgrade Verification
* Remove RHOAM from fresh installation
* Install RHOAM from the `Current Feature` index
  * Ensure the  config image has being reconciled correctly
* Wait for installation is complete. 
* Set up 3scale product
* Set up requests to that product which loop indefinitely
* Login into the customer grafana and confirm requests are showing on the rate limiting dashboard

* Upgrade to the `Updated Version` index. _Note: the upgrade will auto install._
* Wait for installation to complete
* Check the following:
  * Observability operator namespaces are removed
  * Finalizers are removed from the RHMI Cr for the observability operator
  * Status block should be complete for the observability operator.
* Refresh the customer rate limiting dashboard. Data still should be displayed and graph up to date. 

## Regression testing
The work as part of https://github.com/integr8ly/integreatly-operator/pull/2801 was affected by the change here.
It is an edge case and we most manually confirm the solution is still working.

* `Updated Version` of RHOAM is installed and in complete state.
* Open a terminal to watch the redis limits
```bash
watch "oc rsh -n redhat-rhoam-marin3r $(oc get pods -n redhat-rhoam-marin3r -l app=ratelimit -o json | jq -r '.items[1].metadata.name') /bin/sh -c 'wget -qO - http://127.0.0.1:8080/limits/apicast-ratelimit'"
```
* Open another terminal to remote into a rate limit pod
```bash
oc rsh -n redhat-rhoam-marin3r $(oc get pods -n redhat-rhoam-marin3r -l app=ratelimit -o json | jq -r '.items[0].metadata.name') /bin/sh
```
* Create an extra limit
```bash
wget -O- --post-data='{"namespace": "apicast-ratelimit","max_value": 9999,"seconds": 60,"conditions": ["generic_key == slowpath"],"variables": ["generic_key"]}' --header='Content-Type:application/json' http://127.0.0.1:8080/limits
```
* Verify that the limits gets deleted and re-synced back to just 1 limit
  * Number of limits should go from 2 -> 0 -> 1 in the terminal watching the limits

### CODE REVIEW

File: pkg/products/grafana/reconciler.go [hunk 1]
```diff
@@ -404,14 +404,22 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, client k8sclient.C
 
 	r.log.Infof("Grafana CR: ", l.Fields{"status": status})
 
-	observabilityConfig, err := r.ConfigManager.ReadObservability()
-	if err != nil {
-		return integreatlyv1alpha1.PhaseFailed, err
+	if resources.IsInProw(r.installation) {
```
Comment (Patryk-Stefanski) [inline pkg/products/grafana/reconciler.go:407]:
```suggestion
	if !resources.IsInProw(r.installation) {
```

**Comment by Patryk-Stefanski** [comment]
The code looks good, both fresh installation and upgrade installation verification were successful.
/lgtm

**Comment by austincunningham** [comment]
Any reason for the large coverage drop ?

**Comment by Boomatang** [comment]
I really don't understand where is getting that number from. It seems a bit confused. Its possible it is confused with the amount of files and code removed. If anything it possible should have went up as untested code was removed.

**Comment by carlkyrillos** [comment]
This already has an lgtm but I completed the verification steps as well and everything worked perfectly. Also confirmed that the uninstallation is now working on this PR after we rebased the feature branch.

Not sure what's up with the `codecov/project` check - the [summary](https://app.codecov.io/gh/integr8ly/integreatly-operator/pull/3278) doesn't even seem to explain the drop in coverage

**Comment by austincunningham** [comment]
Agree coverall output doesn't make sense in this case , may be due to the amount of covered files being removed, a rebase against the feature branch may improve matters, but I am doubtful. I will approve and override 

/approve
/override codecov/project

**Pull request merged**